LiteCore/Support/LogEncoder.hh

```diff
@@ -68,16 +68,16 @@ namespace litecore {
         void                  _scheduleFlush();
         void                  performScheduledFlush();
 
-        std::mutex                           _mutex;
-        fleece::Writer                       _writer;
-        std::ostream&                        _out;
-        std::unique_ptr<actor::Timer>        _flushTimer;
-        fleece::Stopwatch                    _st;
-        int64_t                              _lastElapsed{0};
-        int64_t                              _lastSaved{0};
-        LogLevel                             _level;
-        std::unordered_map<size_t, unsigned> _formats;
-        std::unordered_set<unsigned>         _seenObjects;
+        std::mutex                                _mutex;
+        fleece::Writer                            _writer;
+        std::ostream&                             _out;
+        std::unique_ptr<actor::Timer>             _flushTimer;
+        fleece::Stopwatch                         _st;
+        int64_t                                   _lastElapsed{0};
+        int64_t                                   _lastSaved{0};
+        LogLevel                                  _level;
+        std::unordered_map<std::string, unsigned> _formats;
```

Comment:
What's the motivation behind changing this from size_t (pointer) to std::string?

Reply:
Ah I see. Is this map inserted to / elements looked up frequently ? String comparison is much more expensive than pointer comparison. Are you concerned with any performance implications, or you think it should be fine? I wish we had some quick perf tests we could run to verify

Reply:
The cost difference is mostly due to the time to compute the hash. It should be linear in terms of the size of the underlying data. I tried the following.
A. creating a map<string, unsigned>  for 10000 strings of char[80]
B. creating a map<size_t, unsigned>  for 10000 size_t cast from char* of above char strings.
The difference is about 4 times, 
elpased time of string map: 0.0590126
elpased time of size_t map: 0.0115147
Maybe we can reduce the cost by making the string map only for those formats that are affected by "Coll=" or "CorrsID=".